File: solutions/python/darts/1/darts.py
from math import hypot

_distance_score: tuple[tuple[int, int], ...] = ((1, 10), (5, 5), (10, 1))

def score(x: int, y: int) -> int:
    radius: Final[float] = hypot(x, y)
    return next((score for distance, score in _distance_score if radius <= distance), 0)

File: solutions/lua/darts/1/darts.lua
return {
  score = function(x, y)
    local radius = math.sqrt(x ^ 2 + y ^ 2)
    if radius > 10 then
      return 0
    elseif radius > 5 then
      return 1
    elseif radius > 1 then
      return 5
    else 
      return 10
    end
  end
}